test/int/adapter.test.ts

```diff
@@ -185,5 +185,38 @@ suite('Chrome Debug Adapter etc', () => {
             // force kill chrome here, as it will be left open by the debug adapter (same behavior as v1)
             killAllChrome();
         });
+
+
+        test('Should launch successfully on port 0', async () => {
+            // browser already launched to the default port, and navigated away from about:blank
+            const remoteDebuggingPort = 0;
+            await Promise.all([
+                dc.configurationSequence(),
+                dc.launch({ url: 'http://localhost:7890', timeout: 2000, webRoot: testProjectRoot, port: remoteDebuggingPort }),
+            ]);
+            // send some commands over CDP to verify the connection
+            const pauseResult = await dc.pauseRequest({threadId: 0});
+            expect(pauseResult.success).to.be.true;
+            const continueResult = await dc.continueRequest();
+            expect(continueResult.success).to.be.true;
+        });
+
+        test('Should launch successfully on port 0, even when a browser instance is already running', async () => {
+            // browser already launched to the default port, and navigated away from about:blank
+            const remoteDebuggingPort = 0;
+            const dataDir = path.join(__dirname, 'testDataDir');
+            const browser = await puppeteer.launch({ headless: false, args: ['http://localhost:7890', `--user-data-dir=${dataDir}`, `--remote-debugging-port=${remoteDebuggingPort}`] });
+            await Promise.all([
+                dc.configurationSequence(),
+                dc.launch({ url: 'http://localhost:7890', timeout: 2000, webRoot: testProjectRoot, port: remoteDebuggingPort, userDataDir: dataDir }),
```

Comment:
I'd launch a different url/page to verify that we connect to the right tab

Reply:
Good point, done